rust/src/push/mod.rs

```diff
@@ -405,7 +405,7 @@ pub struct EventPropertyIsTypeCondition {
     pub key: Cow<'static, str>,
     // During serialization, the pattern_type property gets replaced with a
     // pattern property of the correct value in synapse.push.clientformat.format_push_rules_for_user.
-    pub value_type: Cow<'static, EventMatchPatternType>,
```

Comment:
This doesn't look right, the the code still searches for `value_type` at https://github.com/matrix-org/synapse/pull/15781/files#diff-50b62bc5724dfc17c692c9467f9a1f92af6ecb2b2afdc41cf96939e794d12f59R68

Reply:
I'm not familiar at all with this code, so I might be super wrong. Is this using an internal format that is live-converted to what the spec mandates? If so, then does the rust module use the internal format or the spec format?

Reply:
Yes it is an internal format, the code I linked to is the serialization code which converts it.